src/validate_constraints.c

```diff
@@ -2224,6 +2366,7 @@ cnode_canonify_pass2(struct jvst_cnode *tree)
 				assert(node->u.mcase.tmp == NULL);
 
 				result = jvst_cnode_canonify(node);
+				assert(result != NULL);
```

Comment:
Is this really invariant?

Reply:
Currently it is.  jvst_cnode_canonify will either return something not-NULL or abort().  I think that cnode/IR/op transformations should assume valid input and always return valid output.  They aren't directly called by users.